src/python/pants/core/util_rules/environments.py

```diff
@@ -313,13 +314,36 @@ class RemoteFallbackEnvironmentField(FallbackEnvironmentField):
     )
 
 
+class RemoteEnvironmentCacheBinaryDiscovery(BoolField):
+    alias = "cache_binary_discovery"
+    default = False
+    help = softwrap(
+        f"""
+        If true, will cache system binary discovery, e.g. finding Python interpreters.
+
+        When safe to do, it is preferable to set this option to `true` for faster performance by
+        avoiding wasted work. Otherwise, Pants will research for system binaries whenever the
+        Pants daemon is restarted.
+
+        However, it is only safe to set this to true if the remote execution environment has a
+        stable environment, e.g. the server will not change versions of installed system binaries.
+        Otherwise, you risk caching results that become stale when the server changes its
+        environment, which may break your builds. Often, with remote execution servers you can
```

Comment:
I wouldn't assume that remote execution servers have a "Docker container" capability. Also, it would be correct to say "Docker image to run" not container.